lua/lspconfig/server_configurations/foam.lua

```diff
@@ -0,0 +1,26 @@
+local util = require 'lspconfig.util'
+
+return {
+  default_config = {
+    -- `foam-ls` can be a shell script executing:
+    -- node /path/to/foam-language-server/lib/foam-ls.js --stdio
+    cmd = { 'foam-ls' },
+    filetypes = { 'foam', 'OpenFOAM' },
+    root_dir = util.root_pattern('system/controlDict'),
```

Comment:
I don't believe this is a valid pattern. I guess you want to check for the existence of a `controlDict` dir/file inside a `system` directory? The current root dir detection implementation can only traverse the file system in one direction (upwards)

Reply:
Yes, I want to check a `controlDict` file inside a `system` dir. If this is not possible, how about checking for just `system`? Because somehow that works in my local setup

Reply:
You could also provide your own matcher function, something like this:

```lua
root_dir = function (fname)
    return util.search_ancestors(fname, function (path)
        if util.path.exists(util.path.join(path, "system", "controlDict")) then
           return path
        end
    end)
end
```

Reply:
Thanks! will do that